File: app/src/main/java/com/antonio/samir/meteoritelandingsspots/ui/meteoriteRecyclerView/selector/MeteoriteSelectorFactory.java
package com.antonio.samir.meteoritelandingsspots.ui.meteoriteRecyclerView.selector;

/**
 * MeteoriteSelector factory used by MeteoriteAdapter to prevent it figure out
 * which should be the proper command to execute
 */
public class MeteoriteSelectorFactory {


    public static MeteoriteSelector getMeteoriteSelector(boolean isLandscape, final MeteoriteSelectorView view) {
        final MeteoriteSelector meteoriteSelector;
        if (isLandscape) {
            meteoriteSelector = new MeteoriteSelectorLandscape(view);
        } else {
            meteoriteSelector = new MeteoriteSelectorPortrait(view);
        }
        return meteoriteSelector;
    }
}

File: app/src/main/java/com/antonio/samir/meteoritelandingsspots/features/list/ui/recyclerView/selector/MeteoriteSelectorFactory.kt
package com.antonio.samir.meteoritelandingsspots.features.list.ui.recyclerView.selector

/**
 * MeteoriteSelector factory used by MeteoriteAdapter to prevent it figure out
 * which should be the proper command to execute
 */
object MeteoriteSelectorFactory {


    fun getMeteoriteSelector(isLandscape: Boolean, view: MeteoriteSelectorView): MeteoriteSelector {
        val meteoriteSelector: MeteoriteSelector
        if (isLandscape) {
            meteoriteSelector = MeteoriteSelectorLandscape(view)
        } else {
            meteoriteSelector = MeteoriteSelectorPortrait(view)
        }
        return meteoriteSelector
    }
}
